src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -5322,124 +5258,97 @@ def _set_log_analytics_workspace_extension(cmd, resource_group_name, vm, vm_name
+    identity['type'] = 'SystemAssigned'
+    if updated_user_assigned:
+        identity['type'] = 'SystemAssigned, UserAssigned' if updated_system_assigned else 'UserAssigned'
+        identity['user_assigned_identities'] = dict.fromkeys(updated_user_assigned, {})
 
-    return instance
+    args['identity'] = identity
+    result = LongRunningOperation(cmd.cli_ctx)(_Update(cli_ctx=cmd.cli_ctx)(command_args=args))
+    return result.get('identity', None)
 
 
-def assign_disk_encryption_set_identity(cmd, client, resource_group_name, disk_encryption_set_name,
+def remove_disk_encryption_set_identity(cmd, resource_group_name, disk_encryption_set_name,
                                         mi_system_assigned=None, mi_user_assigned=None):
-    DiskEncryptionSetUpdate, EncryptionSetIdentity = cmd.get_models('DiskEncryptionSetUpdate', 'EncryptionSetIdentity',
-                                                                    operation_group='disk_encryption_sets')
-    from azure.cli.core.commands.arm import assign_identity as assign_identity_helper
-    client = _compute_client_factory(cmd.cli_ctx)
-
-    def getter():
-        return client.disk_encryption_sets.get(resource_group_name, disk_encryption_set_name)
-
-    def setter(disk_encryption_set, mi_system_assigned=mi_system_assigned, mi_user_assigned=mi_user_assigned):
-        IdentityType = cmd.get_models('DiskEncryptionSetIdentityType', operation_group='disk_encryption_sets')
-        existing_system_identity = False
-        existing_user_identities = set()
-        if disk_encryption_set.identity is not None:
-            existing_system_identity = disk_encryption_set.identity.type in [IdentityType.SYSTEM_ASSIGNED_USER_ASSIGNED,
-                                                                             IdentityType.SYSTEM_ASSIGNED]
-            existing_user_identities = {x.lower() for x in
-                                        list((disk_encryption_set.identity.user_assigned_identities or {}).keys())}
-
-        add_system_assigned = mi_system_assigned
-        add_user_assigned = {x.lower() for x in (mi_user_assigned or [])}
-
-        updated_system_assigned = existing_system_identity or add_system_assigned
-        updated_user_assigned = list(existing_user_identities.union(add_user_assigned))
-
-        identity_types, user_assigned_identities = _build_identities_info_from_system_user_assigned(
-            cmd, updated_system_assigned, updated_user_assigned)
-
-        encryption_set_identity = EncryptionSetIdentity(type=identity_types,
-                                                        user_assigned_identities=user_assigned_identities)
-
-        disk_encryption_set_update = DiskEncryptionSetUpdate()
-        disk_encryption_set_update.identity = encryption_set_identity
-        return patch_disk_encryption_set(cmd, resource_group_name, disk_encryption_set_name, disk_encryption_set_update)
+    remove_system_assigned_identity = mi_system_assigned is not None
```

Comment:
Can the value be `False`?
```suggestion
    remove_system_assigned_identity = mi_system_assigned is True
```

Reply:
It's expect to work as `True` when it's not given any value, 
like `az disk-encryption-set identity remove --system-assigned`

Reply:
Can the users input `--system-assigned False`?